### PR TITLE
Explain taxonomy (resolves #10)

### LINKED ISSUE #10: small readme clarifications
I was thinking the Readme could be refined slightly, on a few counts:

1. Some line would be nice that say `plotsbot is an integrated system for bots across various interfaces, such as in an IRC chatroom, in GitHub issues, or on PublicLab.org. The bot consists of a set of behaviors, like "Greet" or "Help" (see below) which can work on one or more interface.`
1. https://github.com/publiclab/plotsbot#chatbot could be added to, clarifying that it's an IRC chat interface?
1. Generally, the taxonomy of "interfaces" vs. "behaviors" could be a great way to organize the README as well as the code. It might lead to questions like "How do I get behavior X to run on interface B?" which would be great illustrations of this architecture.

### PR DESCRIPTION
**Pull request by ryzokuken**

### CODE REVIEW

**Comment by ryzokuken** [comment]
@jywarren @ananyo2012 Please look into this one.

@publiclab/soc feedback appreciated.

File: README.md [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
+1. Private Interfaces where one to one conversations take place between the user and plotsbot. Each message sent by the user is meant for plotsbot and vice versa and should be interpreted as such.
+2. Public Interfaces where many to one conversation takes place between a number of users and plotsbot. Each message sent by a user might or might not be meant for a specific other user (or plotsbot), and thus each message meant for plotsbot should consider its name explicitly.
+
+### IRC
+
+This interface allows plotsbot to interact with users on IRC. In production, plotsbot connects to #publiclab channel on the OFTC IRC network. For testing purposes, technically the bot can be made to connect to any possible channel but #publiclab-testing on OFTC is dedicated towards the purpose of testing the bot.
+
+IRC classifies as both a public and a private interface as the general IRC channel acts as a public interface to the bot while a user can DM the bot which would act as a private interface to the bot.
+
+### CLI
```
Comment (ananyo2012) [inline README.md]: You can add this now.

**Comment by ryzokuken** [comment]
@jywarren Actually, behaviors like `greet` and `help` are appropriately grouped under **Services** for instance, behaviors related to Github would go under the **Github Service** inside `services/github.js`. Do you dislike this model? Perhaps we could segregate each behavior, but I feel strongly that grouping them works well for us.

**Comment by jywarren** [comment]
Is the API for interfaces consistent enough that a behavior could be used
for multiple interfaces? That's what I'm suggesting, here, does that make
sense?

**Comment by ryzokuken** [comment]
That does. As of now atleast (with two interfaces under our belt and a few more to come), the API for interfaces is consistent enough that the behaviors be completely agnostic of which interface is infact running. All interface specific activities as handled inside the modules themselves, freeing the behaviors of any guesswork. Does that clear things up? I am open to clarifying things further.

**Comment by jywarren** [comment]
Yes, so then to me, it makes sense that behaviors are perhaps written with
one interface in mind, but as presented in the readme, we might say "greet"
is "intended for the chat interface" but make it clear that behaviors
should be at least somewhat agnostic. How about that? Each might indicate a
"primary" interface but not note that they should be able to work across
others. I just don't want people to write copies of behaviors for different
interfaces unless it's totally necessary, so i'd like that to come through
in the docs. Thanks!

**Comment by ryzokuken** [comment]
@jywarren You're right. While all behaviors (up until now, atleast) were designed keeping in mind the chat interface for the bot, the universal API for the interfaces allow each behavior or enjoy an interface agnostic status. If there's no clear indication of this fact in the README, I could add a paragraph around the **Services** subheading confirming that each behavior should be interface agnostic and should call the general functions available in each interface specified by an API "specification". This specification would be added to the docs in another PR where I tell users how to suggest/code new interfaces and behaviors.

**Comment by jywarren** [comment]
That sounds good. See how I tried to show the conventions here: https://github.com/publiclab/PublicLab.Editor#modules

Actually that example could be even more explicit. See also https://github.com/publiclab/image-sequencer#contributing-modules -- that's a good one, and you may want to chat with @ccpandhare as that's a similarly well-modularized project. 

Thanks, @ryzokuken! 🎈

**Comment by ryzokuken** [comment]
Read both of those and found it wonderful. Now I have some inspiration when writing the specifications for the interface API.

**Comment by ananyo2012** [comment]
This looks great! I like the present system of modularization. It is be good to have code samples in the readme that explain how one can write a new module be it a new interface or a new service. The PublicLab.Editor project has a good amount of code samples explaining each https://github.com/publiclab/PublicLab.Editor#modules-1 which makes it easier to new contributors to follow.

**Pull request merged**